tree-sitter-darklang/grammar.js

```diff
@@ -139,12 +154,63 @@ module.exports = grammar({
 
     infix_operation: $ =>
       // given `1 + 2 * 3`, this will parse as `1 + (2 * 3)`
-      prec.left(
-        1,
-        seq(
-          field("left", $.expression),
-          field("operator", alias(choice("+", "-"), $.operator)), // TODO more operators.
-          field("right", $.expression), // TODO maybe optional
+      choice(
```

Comment:
I think we should refactor this, into two buckets: the left-associated operators, and the right-associated ones. I think F# does similar. Otherwise, the grammar becomes a bit harder to read (redundant)

Reply:
If I'm wrong about this, and you're doing the standard thing, then please let me know!

Reply:
there's potentially some value in having them all separate like this, because we can be extra-specific about the precedences. Which makes me think that the code might be OK as-is, but wanted to bring up for discussion.

Reply:
I'll actually merge as-is, thinking I'm wrong, but worth some follow-up maybe

Reply:
Yes, all this separation was to handle precedence correctly, but I'm not sure if it's the standard way to do it, so I'll need to verify

Reply:
Nevermind -- this is OK. This embeds the precedence, which is really important. was prematurely thinking to thin down code is all.